afterpay/src/main/kotlin/com/afterpay/android/view/AfterpayPriceBreakdown.kt

```diff
@@ -0,0 +1,150 @@
+import android.text.style.ImageSpan
+import android.text.style.URLSpan
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.Gravity
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.TextView
+import androidx.annotation.AttrRes
+import androidx.annotation.ColorInt
+import androidx.core.content.ContextCompat
+import com.afterpay.android.R
+
+class AfterpayPriceBreakdown(context: Context, attrs: AttributeSet?) : FrameLayout(context, attrs) {
+    constructor(context: Context) : this(context, attrs = null)
+
+    var colorScheme: AfterpayColorScheme = AfterpayColorScheme.DEFAULT
+        set(value) {
+            field = value
+            updateText()
+            invalidate()
+            requestLayout()
+        }
+
+    private val textView: TextView = TextView(context).apply {
+        setTextColor(context.resolveColorAttr(android.R.attr.textColorPrimary))
+        setLineSpacing(0f, 1.2f)
+        textSize = 14f
+        movementMethod = LinkMovementMethod.getInstance()
+        gravity = Gravity.CENTER
+        importantForAccessibility = View.IMPORTANT_FOR_ACCESSIBILITY_NO
+        layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT).apply {
+            gravity = Gravity.CENTER
+        }
+    }
+
+    init {
+        layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT)
+        importantForAccessibility = View.IMPORTANT_FOR_ACCESSIBILITY_YES
+        isFocusable = true
+
+        addView(textView)
+
+        context.theme
+            .obtainStyledAttributes(attrs, R.styleable.Afterpay, 0, 0)
+            .apply {
+                try {
+                    val value = getInteger(R.styleable.Afterpay_colorScheme, 0)
+                    colorScheme = AfterpayColorScheme.values()[value]
+                } finally {
+                    recycle()
+                }
+            }
+
+        updateText()
+    }
+
+    private fun updateText() {
+        val drawable = resources.getDrawable(colorScheme.badgeDrawable, null).apply {
+            val metrics = textView.paint.fontMetrics
+            val fontHeight = metrics.descent - metrics.ascent
+            val aspectRatio = intrinsicWidth / intrinsicHeight.toFloat()
+            val drawableHeight = fontHeight * 2.5
+            val drawableWidth = drawableHeight * aspectRatio
+            setBounds(0, 0, drawableWidth.toInt(), drawableHeight.toInt())
+        }
+
+        textView.apply {
+            text = SpannableStringBuilder()
+                .append(resources.getString(R.string.price_breakdown_total_cost))
+                .append(" ")
+                .append(" ", CenteredImageSpan(drawable), Spannable.SPAN_INCLUSIVE_EXCLUSIVE)
+                .append(" ")
+                .append(
+                    resources.getString(R.string.price_breakdown_info_link),
+                    URLSpan("https://www.afterpay.com"),
```

Comment:
This is actually going to need to be a different URL based on region, the current US one is: https://static-us.afterpay.com/javascript/modal/us_modal.html

However they may be different soon.

Reply:
I believe the original plan was for us to open this up in a modal web view